controlplane/src/db/schema.ts

```diff
@@ -521,3 +521,48 @@ export const gitInstallations = pgTable('git_installations', {
   providerName: text('provider_name').notNull(),
   oauthToken: text('oauth_token'),
 });
+
+export const integrationTypeEnum = pgEnum('integration_type', ['slack']);
+
+export const organizationIntegrations = pgTable('organization_integrations', {
+  id: uuid('id').notNull().primaryKey().defaultRandom(),
+  organizationId: uuid('organization_id')
+    .notNull()
+    .references(() => organizations.id, {
+      onDelete: 'cascade',
+    }),
+  name: text('name').notNull(),
+  events: text('events').array(),
+  type: integrationTypeEnum('type').notNull(),
+  createdAt: timestamp('created_at', { withTimezone: true }).notNull().defaultNow(),
+});
+
+export const slackIntegrationConfigs = pgTable('slack_integration_configs', {
+  id: uuid('id').notNull().primaryKey().defaultRandom(),
+  integrationId: uuid('integration_id')
+    .notNull()
+    .references(() => organizationIntegrations.id, {
+      onDelete: 'cascade',
+    }),
+  endpoint: text('endpoint').notNull(),
+});
+
+export const slackIntegrationEventConfigs = pgTable('slack_integration_event_configs', {
+  id: uuid('id').notNull().primaryKey().defaultRandom(),
+  slackIntegrationConfigId: uuid('slack_integration_config_id')
+    .notNull()
+    .references(() => slackIntegrationConfigs.id, {
+      onDelete: 'cascade',
+    }),
+  event: text('event').notNull(),
+  federatedGraphIds: uuid('federated_graph_ids').array(),
```

Comment:
Could you do it similarly to webhooks? We maintain a table to store the relations. In that way, we don't have to maintain consistency and use classic joins to get all the data.

Reply:
Done